recipes/googleapis/all/conanfile.py

```diff
@@ -11,18 +17,122 @@ class GoogleAPIS(ConanFile):
     homepage = "https://github.com/googleapis/googleapis"
     topics = "google", "protos", "api"
     settings = "os", "arch", "compiler", "build_type"
-    short_paths = True
+    generators = "cmake", "cmake_find_package"
+    options = {
+        "shared": [True, False], 
+        "fPIC": [True, False]
+        }
+    default_options = {
+        "shared": False, 
+        "fPIC": True
+        }
+    exports = "helpers.py"
+
+    def layout(self):
+        cmake_layout(self)
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
 
     def source(self):
         get(self, **self.conan_data["sources"][str(self.version)], destination=self.source_folder, strip_root=True)
 
-    def package_id(self):
-        self.info.header_only()
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+
+    def requirements(self):
+        self.requires('protobuf/3.21.1')
+
+    @property
+    def _cmake_new_enough(self):
+        try:
+            import re
+            output = StringIO()
+            self.run("cmake --version", output=output)
+            m = re.search(r'cmake version (\d+)\.(\d+)\.(\d+)', output.getvalue())
+            major, minor = int(m.group(1)), int(m.group(2))
+            assert major >= 3 and minor >= 20
+        except:
+            return False
+        else:
+            return True
+
+    def build_requirements(self):
+        self.build_requires('protobuf/3.21.1')
+        # CMake >= 3.20 is required. There is a proto with dots in the name 'k8s.min.proto' and CMake fails to generate project files
+        if not self._cmake_new_enough:
+            self.build_requires('cmake/3.23.2')
```

Comment:
Set unconditionally and simplify?

Reply:
I think thats a fair choice, it's fairly complicated code